### DISCUSSION TITLE
Toggle Ligatures With Keybind

### DISCUSSION THREAD

**Discussion by jdholtz** [kovidgoyal/kitty#6133] [Q&A] [answered]
Hi. Is there anyway to toggle ligatures with one keybind? Currently, I have binded one key to disable ligatures and one key to enable ligatures:
```
map alt+1 disable_ligatures_in all always
map alt+2 disable_ligatures_in all never
```

I have looked in the configuration documentation but nothing appears to give this functionality.

If a feature for this doesn't exist, I'd be interested in implementing this as I think it would be helpful to be able to toggle this instead of having to press two separate keys for this functionality.

**Comment by page-down** [comment, marked as answer]
> I have looked in the configuration documentation ...
> ...
> ... I'd be interested in implementing this ...

You can implement this by writing a custom kitten that checks the current state and performs the relevant action.

https://sw.kovidgoyal.net/kitty/kittens/custom/

**Reply by jdholtz** [reply]
I was not aware of this. Thanks!
